torii/src/routing.rs

```diff
@@ -151,15 +152,19 @@ pub async fn handle_pending_transactions(
     queue: Arc<Queue>,
     sumeragi: SumeragiHandle,
     pagination: Pagination,
-) -> Result<Scale<Vec<SignedTransaction>>> {
+    payload: TransactionPayload,
+) -> Result<Scale<Option<SignedTransaction>>> {
     let query_response = sumeragi.apply_wsv(|wsv| {
         queue
             .all_transactions(wsv)
             .map(Into::into)
             .paginate(pagination)
-            .collect::<Vec<_>>()
-        // TODO:
-        //.batched(fetch_size)
+            .find(|current_transaction: &SignedTransaction| {
```

Comment:
This is much better than scanning all the transaction on client, but this is still doing a linear search of pending transactions. This might get slow in case there a lot of pending transactions.

Do we expect a large (>100) number of pending transactions? If yes, we can maintain a `HashMap` to speed up this search

Reply:
I do not have an answer regarding the size of the queue, although it should be important that it is indeed a queue that maintains an order. So we need some kind of `HashMap` that remembers insertion ordering.
I think that the comparison function itself is kinda slow (it compares strings in `AccountId`) and I don't really see any speed gain on such a small number of transactions (if it's indeed in range of 100-1000 or less). is there anyone who can help?

Reply:
you can make a wrapper over a tranasction and implement custom `Hash` and `PartialEq` and `PartialOrd` over it so that they exclude `creation_time`. Then you can put it into a `HashMap`. We don't need insertion ordering since transactions are guaranteed to be unique in the `queue`